### PR TITLE
#106 - type parameter for overapproximate/decompose

### PR DESCRIPTION
**Pull request by schillic**
See #106.

### CODE REVIEW

File: test/unit_decompose.jl [hunk 1]
```diff
@@ -9,7 +9,7 @@ Y = [1.92664 1.00674 1.0731 -0.995149;
     -2.43723 0.782825 -3.99255 3.93324]
 Y = Y * CPA([BallInf([0.767292, 0.936613], 0.1), BallInf([0.734104, 0.87296], 0.1)])
 Ω0 = CH(X, Y)
-dec = Approximations.decompose(Ω0)
+dec = Approximations.decompose(Ω0, HPolygon)
```
Comment (mforets) [inline test/unit_decompose.jl]: can you make the second argument optional?

File: src/Approximations/2D_approximations.jl [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+### Input
+
+- `S` -- convex set, assumed to be two-dimensional
+- `HPolygon` for dispatch
+
+### Output
+
+A box-shaped polygon in constraint representation.
+"""
+function overapproximate(S::LazySet, ::Type{HPolygon})::HPolygon
```
Comment (mforets) [inline src/Approximations/2D_approximations.jl]: i don't understand, where is the second argument used?

**Comment by mforets** [comment]
ok, i see. i would suggest to make this change an optional keyword argument, eg. `"type"` or `"set_type"`, being `HPolygon` the default one.

File: src/Approximations/2D_approximations.jl (hunk 2 above)
Comment (mforets) [reply src/Approximations/2D_approximations.jl]: nevermind. i think i what i wanted to suggest is defining

```julia
overapproximate(S::LazySet) = overapproximate(S::LazySet, ::Type{HPolygon})
```

Comment (schillic) [reply src/Approximations/2D_approximations.jl]: > where is the second argument used?

It's only used by Julia to find the correct method.

Comment (schillic) [reply src/Approximations/2D_approximations.jl]: > maybe define
> `overapproximate(S::LazySet) = overapproximate(S, HPolygon)`

Good idea, I will try.

File: test/unit_decompose.jl (hunk 1 above)
Comment (schillic) [reply test/unit_decompose.jl]: Okay, same as below.

**Comment by schillic** [comment]
I rebased with `master` and incorporated your suggestion with the default value. This also saved some changes to unit tests and plot recipes 👍

**Comment by mforets** [comment]
failing checks are a Travis CI timeout => positive review

**Pull request merged**